### PR TITLE
Add NoncontiguousPartitioner::temporary_storage_size()

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

**Review by drwells — APPROVED** [review]
I remember seeing that this was the correct value in the debugger - I don't fully understand the indexing scheme but this should be correct.

File: include/deal.II/base/mpi_noncontiguous_partitioner.templates.h [hunk 1]
```diff
@@ -54,12 +54,19 @@ namespace Utilities
...
 
     std::pair<unsigned int, unsigned int>
-    NoncontiguousPartitioner::n_targets()
+    NoncontiguousPartitioner::n_targets() const
     {
       return {send_ranks.size(), recv_ranks.size()};
     }
 
 
+    unsigned int
```
Comment (kronbichler) [inline include/deal.II/base/mpi_noncontiguous_partitioner.templates.h:63]: Can we add a third empty line above this?

File: include/deal.II/base/mpi_noncontiguous_partitioner.templates.h [hunk 2]
```diff
@@ -252,7 +259,8 @@ namespace Utilities
         requests.resize(send_ranks.size() + recv_ranks.size());
 
       if (this->buffers.size() != send_ptr.back() * sizeof(Number))
-        this->buffers.resize(send_ptr.back() * sizeof(Number), 0);
+        this->buffers.resize(this->temporary_storage_size() * sizeof(Number),
+                             0);
```
Comment (kronbichler) [inline include/deal.II/base/mpi_noncontiguous_partitioner.templates.h:263]: Do we need to set the zero or leads relying on the default value to easier code?

**Review by kronbichler — APPROVED** [review]

File: include/deal.II/base/mpi_noncontiguous_partitioner.templates.h (hunk 2 above)
Comment (peterrum) [reply include/deal.II/base/mpi_noncontiguous_partitioner.templates.h]: This is not needed. I have remove it.

File: include/deal.II/base/mpi_noncontiguous_partitioner.templates.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/base/mpi_noncontiguous_partitioner.templates.h:64]: Done.

**Comment by drwells** [comment]
All changes were made so I'll merge :)

**Pull request merged**